Task3/src/Repositories/FilterApplier.hs

```diff
@@ -0,0 +1,18 @@
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+
+module Repositories.FilterApplier 
+    ( applyFilter
+    , applyPagination) where
+import Data.SearchModels (SearchModel(..))
+
+applyFilter ::(SearchModel c) => (a -> b) -> (c -> Maybe b) -> (b -> b -> Bool) -> c -> [a] -> [a]
+applyFilter arrSelect searchModelSelect funcfilter searchModel arr =
+    let maybeSearchValue = searchModelSelect searchModel
+    in searchInArr maybeSearchValue
+    where
+        searchInArr (Just value) = filter (funcfilter value . arrSelect) arr
```

Comment:
SO, you actually have this use case I wanted you to encounter, you just didn't add a function definition yet. Please add types definition for your searchInArr function.